wandb/old/summary.py

```diff
@@ -390,6 +390,10 @@ def __init__(self, run, client, summary=None):
         self._client = client
         self._started = time.time()
 
+    def __delitem__(self, key):
+        if key in self._json_dict:
+            del self._json_dict[key]
```

Comment:
You likely want to mimic the behavior of a regular dict that throws an error if the key is not present, something like
```suggestion
        if key not in self._json_dict:
            raise KeyError(key)
        del self._json_dict[key]
```